src/client/datascience/errors/errorRendererComms.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { commands, notebooks, Position, Range, Selection, TextEditorRevealType, Uri } from 'vscode';
+import { IExtensionSyncActivationService } from '../../activation/types';
+import { IApplicationShell, ICommandManager, IDocumentManager } from '../../common/application/types';
+import { IFileSystem } from '../../common/platform/types';
+import { IDisposableRegistry } from '../../common/types';
+import { InteractiveWindowMessages } from '../interactive-common/interactiveWindowTypes';
+import { LineQueryRegex, linkCommandAllowList } from '../interactive-common/linkProvider';
+
+@injectable()
+export class ErrorRendererCommunicationHandler implements IExtensionSyncActivationService {
+    constructor(
+        @inject(IDisposableRegistry) private readonly disposables: IDisposableRegistry,
+        @inject(IDocumentManager) private readonly documentManager: IDocumentManager,
+        @inject(ICommandManager) private readonly commandManager: ICommandManager,
+        @inject(IFileSystem) private readonly fs: IFileSystem,
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell
+    ) {}
+
+    activate(): void {
```

Comment:
Moved out of interactive window class, this is not specific to just interactive window anymore.